layout-server/src/test/java/org/finos/vuu/layoutserver/integration/LayoutIntegrationTest.java

```diff
@@ -0,0 +1,389 @@
+        String request = "invalidRequest";
+
+        mockMvc.perform(put("/layouts/{id}", layout.getId())
+                .content(objectMapper.writeValueAsString(request))
+                .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andExpect(content().string(
+                "JSON parse error: Cannot construct instance of `org.finos.vuu.layoutserver.dto"
+                    + ".request.LayoutRequestDTO` (although at least one Creator exists): no "
+                    + "String-argument constructor/factory method to deserialize from String "
+                    + "value ('invalidRequest'); nested exception is com.fasterxml.jackson"
+                    + ".databind.exc.MismatchedInputException: Cannot construct instance of `org"
+                    + ".finos.vuu.layoutserver.dto.request.LayoutRequestDTO` (although at least "
+                    + "one Creator exists): no String-argument constructor/factory method to "
+                    + "deserialize from String value ('invalidRequest')\n"
+                    + " at [Source: (org.springframework.util.StreamUtils$NonClosingInputStream);"
+                    + " line: 1, column: 1]"));
+
+        assertThat(layoutRepository.findById(layout.getId()).orElseThrow()).isEqualTo(
+            layout);
+    }
+
+    @Test
+    void updateLayout_validIdButLayoutDoesNotExist_returnsNotFound() throws Exception {
+        UUID layoutID = UUID.randomUUID();
+        LayoutRequestDTO layoutRequest = createValidLayoutRequest();
+
+        mockMvc.perform(put("/layouts/{id}", layoutID)
+                .content(objectMapper.writeValueAsString(layoutRequest))
+                .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    void updateLayout_invalidId_returns400() throws Exception {
+        String layoutID = "invalidUUID";
+        LayoutRequestDTO layoutRequest = createValidLayoutRequest();
+
+        mockMvc.perform(put("/layouts/{id}", layoutID)
+                .content(objectMapper.writeValueAsString(layoutRequest))
+                .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest())
+            .andExpect(content().string(
+                "Failed to convert value of type 'java.lang.String' to required type 'java.util"
+                    + ".UUID'; nested exception is java.lang.IllegalArgumentException: Invalid "
+                    + "UUID string: invalidUUID"));
+    }
+
+    @Test
+    void deleteLayout_validIdLayoutExists_returnsSuccessAndLayoutIsDeleted() throws Exception {
+        Layout layout = createDefaultLayoutInDatabase();
+
+        mockMvc.perform(get("/layouts/{id}", layout.getId())).andExpect(status().isOk());
+
+        mockMvc.perform(delete("/layouts/{id}", layout.getId())).andExpect(status().isNoContent());
+
+        mockMvc.perform(get("/layouts/{id}", layout.getId())).andExpect(status().isNotFound());
```

Comment:
I would prefer to just call `mockMvc.perform` for behaviour under test, i.e. hitting the `delete` endpoint. We could do a simpler assertion using the repo instead of the gets before and after.

Reply:
Done! I can't remember in which commit though...